File: problem-9.py
### Problem 9
###-----------------------------------------------------------------------------------------------
### A Pythagorean triplet is a set of three natural numbers, a < b < c, for which, a^2 + b^2 = c^2
### For example, 3^2 + 4^2 = 9 + 16 = 25 = 52.
### There exists exactly one Pythagorean triplet for which a + b + c = 1000.
### Find the product abc.

### Solution

product = 0
for a in range(1, 1001):
	for b in range(a, 1001):
		c = 1000 - a - b
		if (a**2 + b**2) == c**2 and (a + b + c) == 1000:
			product = a*b*c
			break

print(product)

File: java/problem-9.java
public class Problem9 {
    
    public static void main(String[] args) {
        int a = 0;
        int b = 0;
        int c = 0;

        for (int i = 1; i < 1000; i++) {
            for (int j = 1; j < 1000; j++) {
                for (int k = 1; k < 1000; k++) {
                    if (i*i + j*j == k*k) {
                        if (i + j + k == 1000) {
                            a = i;
                            b = j;
                            c = k;
                        }
                    }
                }
            }
        }

        System.out.println("The Pythaogorean Triple is: (" + a + "," +  b + "," + c + ")");
        long product = a*b*c;
        System.out.println("The product of the three is: " + product);
    }

}
